sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Utils.scala

```diff
@@ -151,6 +153,27 @@ private[sql] object DataSourceV2Utils extends Logging {
     }
   }
 
+  /**
+   * Returns the table provider for the given format, or None if it cannot be found.
+   */
+  def getTableProvider(provider: String, conf: SQLConf): Option[TableProvider] = {
+    // Return earlier since `lookupDataSourceV2` may fail to resolve provider "hive" to
+    // `HiveFileFormat`, when running tests in sql/core.
+    if (DDLUtils.isHiveTable(Some(provider))) return None
+    DataSource.lookupDataSourceV2(provider, conf) match {
+      // TODO(SPARK-28396): Currently file source v2 can't work with tables.
+      case Some(_: FileDataSourceV2) => None
+      case o => o
+    }
+  }
+
+  /**
+   * Check if the provider is a v2 provider.
+   */
+  def isV2Provider(provider: String, conf: SQLConf): Boolean = {
```

Comment:
Because `isV2Provider` is only used for  `ResolveSessionCatalog`, move this back to `ResolveSessionCatalog`.

Reply:
Good catch!